hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java

```diff
@@ -48,29 +51,38 @@
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final TokenStore store;
+	private final TokenStore commonStore;
+	private final UniqueStore uniqueStore;
 	private final TransactionContext txnCtx;
 
 	public TokenMintTransitionLogic(
-			TokenStore store,
+			TokenStore commonStore,
+			UniqueStore uniqueStore,
 			TransactionContext txnCtx
 	) {
-		this.store = store;
+		this.commonStore = commonStore;
+		this.uniqueStore = uniqueStore;
 		this.txnCtx = txnCtx;
 	}
 
 	@Override
 	public void doStateTransition() {
 		try {
 			var op = txnCtx.accessor().getTxn().getTokenMint();
-			var id = store.resolve(op.getToken());
+			var id = commonStore.resolve(op.getToken());
 			if (id == TokenStore.MISSING_TOKEN) {
 				txnCtx.setStatus(INVALID_TOKEN_ID);
 			} else {
-				var outcome = store.mint(id, op.getAmount());
+				var token = commonStore.get(id);
+				ResponseCodeEnum outcome;
+				if(token.tokenType().equals(TokenType.FUNGIBLE_COMMON)) {
+					outcome = commonStore.mint(id, op.getAmount());
+				} else {
+					outcome = uniqueStore.mint(id, op.getMetadata().toStringUtf8(), RichInstant.fromJava(txnCtx.consensusTime()));
+				}
 				txnCtx.setStatus((outcome == OK) ? SUCCESS : outcome);
-				if(outcome == OK) {
-					txnCtx.setNewTotalSupply(store.get(id).totalSupply());
+				if (outcome == OK) {
+					txnCtx.setNewTotalSupply(commonStore.get(id).totalSupply());
```

Comment:
In `txnCtx` we have to add new method `setCreated(int64 serialNumber)`. Based on the HAPI spec, the newly created serial number should be added in the Transaction Receipt

Reply:
Can we add it on another PR? I think this is kinda related to batch txs